middle_end/flambda/simplify/simplify_variadic_primitive.ml

```diff
@@ -72,6 +72,51 @@ let simplify_make_block_of_values dacc _prim dbg tag ~shape
     Simplified_named.reachable term, env_extension, args, dacc
   end
 
+let simplify_make_block_of_floats dacc _prim dbg
+      ~(mutable_or_immutable : Mutability.t)
+      args_with_tys ~result_var =
+  let denv = DA.denv dacc in
+  let args, _arg_tys = List.split args_with_tys in
+  let invalid () =
+    let ty = T.bottom K.value in
+    let env_extension = TEE.one_equation (Name.var result_var) ty in
+    Simplified_named.invalid (), env_extension, args, dacc
+  in
+  (* CR mshinwell: This could probably be done more neatly. *)
+  let found_bottom = ref false in
+  let fields =
+    List.map (fun ((arg : Simple.t), arg_ty) ->
+        if T.is_bottom (DE.typing_env denv) arg_ty then begin
```

Comment:
Please add a CR comment here that we should review all similar pieces of code in the file and aim to remove the `T.is_bottom` checks (kind of like #336 did in the simplifier).

Reply:
done